pisek/env/select_solutions.py

```diff
@@ -0,0 +1,66 @@
+# pisek  - Tool for developing tasks for programming competitions.
+#
+# Copyright (c)   2023        Daniel Skýpala <daniel@honza.info>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# any later version.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+from typing import Optional
+
+from pisek.utils.text import tab
+from pisek.env.config_errors import TaskConfigError
+from pisek.env.task_config import TaskConfig, SolutionConfig
+
+
+class UnknownSolutions(TaskConfigError):
+    def __init__(
+        self, unknown: list[str], solutions: dict[str, "SolutionConfig"]
+    ) -> None:
+        def format_solution(s: SolutionConfig) -> str:
+            if s.name == s.source:
+                return s.name
+            else:
+                return f"{s.name} ({s.source})"
```

Comment:
I would try to use something more descriptive than simple ``at``. Maybe like this?
```suggestion
                return f"{s.name} (source: {s.source})"
```

Reply:
That would be clearer. It would make the message slightly longer, but that's probably not an issue, since users are unlikely to pass several incorrect solution names.

Reply:
> That would be clearer. It would make the message slightly longer, but that's probably not an issue, since users are unlikely to pass several incorrect solution names.

We display this for all real solutions, not incorectly named ones.

Reply:
But each of those is on it's own line, so that shouldn't matter, right?

Reply:
Yup.

Reply:
In that case the `"(source: {})"` is probably better.